Makefile.am

```diff
@@ -8,23 +8,23 @@ DIST_SUBDIRS = @TOP_DIST_SUBDIRS@
 
 $(GT_TARGETS_CLEAN):
 	@if [ -f $($@_stamp) ]; then \
-		export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) clean ; \
+		export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) clean || exit 1 ; \
 	fi
 	@rm -f $($@_stamp)
 $(GT_ALL_TARGETS_INSTALL):
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) install
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) install || exit 1
 $(GT_TARGETS_CHECK):
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) check
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && $(MAKE) $(AM_MAKEFLAGS) check || exit 1
 $(GT_ALL_TARGETS_DIST):
-	@$(MKDIR_P) package-output
-	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && MAKEFLAGS= $(MAKE) $(AM_MAKEFLAGS) dist
-	@cp $($@_SUBDIR)/$($@-tarball) package-output
+	@export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) ; cd $($@_SUBDIR) && MAKEFLAGS= $(MAKE) $(AM_MAKEFLAGS) dist && \
+	$(MKDIR_P) package-output && \
+	cp $($@_SUBDIR)/$($@-tarball) package-output || exit 1
```

Comment:
@msalle 
Resolved that too early.

The order of the commands is still different to the original code:

As the `cd in the first part (originally in the second line) changes `$PWD`, the `mkdir` in the second part (originally in the first line) is done in a different directory.

Why change the order?

Reply:
I prefer to do the mkdir only after the make, since it doesn't leave empty directories in case of failures in the make process, but you're right something is still odd there, however the original is also wrong I think: it's not doing a `cd ..`, so how can the cp succeed, since we're still in the `$@_SUBDIR`, or is there some magic going on that I don't see?.
Also, how sure are we that `$@_SUBDIR` is only one level deeper? If so, we can just add a `cd ..` before the mkdir. 
I cannot really experiment right now, probably tomorrow. If you have a chance to figure this one out, I'll push an update (or you can just make a new PR instead).

Reply:
Sorry, no time tomorrow, but can have a look into that next week.

Reply:
@msalle:

Short update:
I seemed to remember having dealt with `make dist` in the past and just had a look into the scripts @matyasselmeci created for Travis CI. Some time ago I added a change to also create the full source installer tarball for the GCT and looking into https://github.com/gridcf/gct/blob/master/travis-ci/make_source_tarballs.sh#L42..L45, the `cp` operation in the Makefile indeed doesn't work as expected, because otherwise the source installer tarball wouldn't have to be manually copied to the `package-output` dir in the shell script.

Reply:
Ok, I figured it out I think. The make automatically jumps back to the top_srcdir. I've now split the setup in two, and let both of them error out.
E.g. `make globus_common-dist` now correctly works again.
Sorry for all the mess )-: